core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -185,33 +196,40 @@ private WeekActivityOverviewDto getWeekActivityOverview(UUID userAnonymizedId, L
 				.collect(Collectors.toSet());
 	}
 
-	private WeekActivityDto createAndSaveWeekInactivity(UserAnonymizedDto userAnonymized, Goal goal, ZonedDateTime startOfWeek,
-			LevelOfDetail levelOfDetail, Set<IntervalInactivityDto> missingInactivities)
+	private WeekActivityDto createAndSaveWeekInactivity(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate,
+			Goal goal, ZonedDateTime startOfWeek, LevelOfDetail levelOfDetail, Set<IntervalInactivityDto> missingInactivities)
 	{
-		return WeekActivityDto.createInstanceInactivity(userAnonymized, goal, startOfWeek, levelOfDetail, missingInactivities);
+		return WeekActivityDto.createInstanceInactivity(userAnonymized, earliestPossibleDate, goal, startOfWeek, levelOfDetail,
+				missingInactivities);
 	}
 
-	private void createAndSaveInactivityDays(UserAnonymizedDto userAnonymized, Set<GoalDto> goals, WeekActivityDto weekActivity,
-			Set<IntervalInactivityDto> missingInactivities)
+	private void createAndSaveInactivityDays(UserAnonymizedDto userAnonymized, LocalDate earliestPossibleDate, Set<GoalDto> goals,
+			WeekActivityDto weekActivity, Set<IntervalInactivityDto> missingInactivities)
 	{
-		weekActivity.createRequiredInactivityDays(userAnonymized, goals, LevelOfDetail.WEEK_OVERVIEW, missingInactivities);
+		weekActivity.createRequiredInactivityDays(userAnonymized, earliestPossibleDate, goals, LevelOfDetail.WEEK_OVERVIEW,
+				missingInactivities);
 	}
 
-	private long getTotalPageableItems(Set<BuddyDto> buddies, ChronoUnit timeUnit)
+	private long getTotalPageableItems(UserAnonymizedDto userAnonymized, Set<BuddyDto> buddies, LocalDate userCreationDate,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "userCreationDate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1172)